ERCS/erc-7730.md

```diff
@@ -0,0 +1,442 @@
+
+References to values in the container
+* `@.amount` refers to the enclosing transaction native currency amount
+
+#### Organizing files
+
+Smartcontracts and EIP 712 messages are often re-using common interfaces or types that share similar display formatting. This specification supports a basic inclusion mechanism that enables sharing files describing specific interfaces or types.
+
+The `includes` top-level key is an array of URLs, pointing to files that MUST follow this same specification.
+
+A wallet using a file including other files SHOULD merge those files into the top-level file. When merging, conflicts for unique keys are resolved by prioritizing the including file first, then by each included file in the order they appear in the including file.
+
+It is NOT RECOMMENDED to include files beyond one level of depth, to avoid descriptions that would be too complex to handle for some wallets.
+
+*Example*
+
+This file defines a generic ERC20 interface for a single approve function:
+```json
+{
+    "context": {
+        "contract" : {
+            "abi": [
+                {
+                    "inputs": [
+                        {
+                            "name": "_spender",
+                            "type": "address"
+                        },
+                        {
+                            "name": "_value",
+                            "type": "uint256"
+                        }
+                    ],
+                    "name": "approve",
+                    "type": "function"
+                }
+            ]
+        }
+    },
+
+    "display": {
+        "formats": {
+            "0x095ea7b3": {
+                "$id" : "approve",
+                "intent": "Approve",
+                "fields": {
+                    "_spender" : {
+                        "label": "Spender",
+                        "format": "addressOrName"
+                    },
+                    "_value" : {
+                        "label": "Amount",
+                        "format": "allowanceAmount",
+                        "params": {
+                            "tokenPath": "$.context.contract.address",
+                            "threshold": "0x8000000000000000000000000000000000000000000000000000000000000000"
+                        }
+                    }
+                },
+                "required": ["_spender", "_value"]
+            }
+
+        }
+    }
+}
+```
+Note that there are no keys for binding the contract to a specific address or owner, nor any contract specific metadata.
+
+The following file would include this generic interface and bind it to the specific USDT contract:
+```json
+{
+    "context": {
+        "$id": "Tether USD",
+        "contract" : {
+            "chainId": 1,
+            "address": "0xdAC17F958D2ee523a2206206994597C13D831ec7"
+        }
+    }, 
+    "includes": ["../examples/clear-signing-example-erc20.json"],
```

Comment:
Allowing to include more than one file per context adds complexity when merging. Is this necessary or could it be limited to a single file include per context? What would be a use-case of including more than one file?

Reply:
Didn't find a good example that would require more than one include for now, so definitely could be limited until we have concrete examples when it would be useful